### PR TITLE
Add MultipleChoiceQuestion.

### PR DESCRIPTION
**Pull request by dzhuang**
Based on ``ChoiceQuestion``, add an optional attribute ``full_match``

Optional. ``True`` or ``False``. If true (default), only answers which fully matches the correct answer is correct.  If false, answers with subset of correct choices will get a proportional credit.

### CODE REVIEW

File: course/page/choice.py [hunk 1]
```diff
@@ -232,7 +240,135 @@ def normalized_answer(self, page_context, page_data, answer_data):
         return self.process_choice_string(
                 page_context,
                 self.page_desc.choices[permutation[choice]])
+# }}}
+
+# {{{ multiple choice question
+
+class MultipleChoiceQuestion(ChoiceQuestion):
+    """
+    A page asking the participant to choose one of multiple answers.
```
Comment (inducer) [inline course/page/choice.py]: That description doesn't quite match what I imagine this does.

File: course/page/choice.py [hunk 2]
```diff
@@ -45,6 +45,14 @@ def __init__(self, field, *args, **kwargs):
         # Translators: "choice" in Choice Answer Form in a choice question.
         self.fields["choice"].label = _("Choice")
 
+class MultipleChoiceAnswerForm(StyledForm):
+    def __init__(self, field, *args, **kwargs):
+        super(MultipleChoiceAnswerForm, self).__init__(*args, **kwargs)
+
+        self.fields["choice"] = field
+        # Translators: "Multiple choice" in Choice Answer Form in a mutliple choice question.
+        self.fields["choice"].label = _("Multiple Choices")
```
Comment (inducer) [inline course/page/choice.py]: I'd perhaps only call this "Choices".

**Comment by inducer** [comment]
This looks great. I'll go ahead and fix the two niggles that I found as well as add a test to the test flow in the sample content.

**Comment by dzhuang** [comment]
I forget to modify  the descriptions :-). 

BTW, I'm not sure whether other descriptions, along with the new attribute name ``full_match``, make sense in English. PLS correct those expressions, thx.

**Comment by inducer** [comment]
I've made a bunch of changes in 79cd1ff: In particular, I've changed how partial credit scoring works and how it is enabled.

**Pull request merged**